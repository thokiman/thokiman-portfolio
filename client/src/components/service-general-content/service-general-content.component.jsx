import React from "react";

import {
  ServiceDescription,
  ServiceDescriptionContainer,
  ServiceDescriptionItem,
  ServiceDescriptionTitle,
  ServiceGeneralContentContainer,
  ServiceTitle,
} from "./service-general-content.styles";

const ServiceGeneralContent = ({ title, description }) => {
  return (
    <ServiceGeneralContentContainer>
      <ServiceTitle>{title}</ServiceTitle>
      <ServiceDescription>
        {description.map(({ id, title, description }) => {
          return (
            <ServiceDescriptionContainer key={id}>
              <ServiceDescriptionTitle>{title}</ServiceDescriptionTitle>
              <ServiceDescriptionItem>{description}</ServiceDescriptionItem>
            </ServiceDescriptionContainer>
          );
        })}
      </ServiceDescription>
    </ServiceGeneralContentContainer>
  );
};

export default ServiceGeneralContent;
