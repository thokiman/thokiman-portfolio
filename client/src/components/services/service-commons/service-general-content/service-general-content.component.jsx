import React from 'react';
import {
  ServiceDescription,
  ServiceDescriptionContainer,
  ServiceDescriptionItem,
  ServiceDescriptionTitle,
  ServiceGeneralContentContainer,
  ServiceTitle,
} from './service-general-content.styles';

export const ServiceGeneralContent = ({ title, description }) => {
  return (
    <ServiceGeneralContentContainer>
      <ServiceTitle className='service-general-content-title'>
        {title}
      </ServiceTitle>
      <ServiceDescription>
        {description.map(({ id, title, description }) => {
          return (
            <ServiceDescriptionContainer key={id}>
              <ServiceDescriptionTitle className='service-general-content-description'>
                {title}
              </ServiceDescriptionTitle>
              <ServiceDescriptionItem className='service-general-content-description'>
                {description}
              </ServiceDescriptionItem>
            </ServiceDescriptionContainer>
          );
        })}
      </ServiceDescription>
    </ServiceGeneralContentContainer>
  );
};

export default ServiceGeneralContent;
