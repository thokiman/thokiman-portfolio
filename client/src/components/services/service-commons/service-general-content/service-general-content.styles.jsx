import styled, { keyframes } from 'styled-components';
const textColorServiceDescriptionTitle = ' rgba(188, 207, 213, 0.9)';
const textColorServiceDescriptionItem = ' rgba(227, 225, 228, 0.9)';
const serviceTitleAnimationColor0 = 'rgba(116,198,212,0.8)';
const whiteSmokeColor = '#e3e1e4';
const maxWidthSmartphone = 800;

export const ServiceGeneralContentContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
`;

const serviceTitle = keyframes`
             0% {
                 color: ${serviceTitleAnimationColor0};
             }

             100% {
                 color:${whiteSmokeColor};
             } 

`;

export const ServiceTitle = styled.div`
  font-family: futura-demi;
  font-size: 8vw;
  margin-bottom: 20px;
  width: 100%;
  height: 100%;
  color: ${whiteSmokeColor};
  letter-spacing: 0.1em;
  font-kerning: auto;
  animation: ${serviceTitle} 3s linear infinite alternate both;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 8vw;
  }
`;

export const ServiceDescription = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  margin-bottom: 20px;
  width: 94%;
  height: 100%;
`;

export const ServiceDescriptionContainer = styled.div`
  margin-bottom: 15px;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;
export const ServiceDescriptionTitle = styled.div`
  font-family: futura-demi;
  font-size: 4vw;
  margin-bottom: 15px;
  text-indent: 0.6em;
  color: ${textColorServiceDescriptionTitle};
  letter-spacing: 0.02em;
  font-kerning: auto;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 6vw;
  }
`;
export const ServiceDescriptionItem = styled.div`
  font-size: 2.5vw;
  text-align: left;
  color: ${textColorServiceDescriptionItem};
  letter-spacing: 0.06em;
  line-height: 1.5em;
  font-kerning: auto;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 3.7vw;
    text-align: left;
  }
`;

ServiceGeneralContentContainer.displayName = 'ServiceGeneralContentContainer';
ServiceTitle.displayName = 'ServiceTitle';
ServiceDescription.displayName = 'ServiceDescription';
ServiceDescriptionContainer.displayName = 'ServiceDescriptionContainer';
ServiceDescriptionTitle.displayName = 'ServiceDescriptionTitle';
ServiceDescriptionItem.displayName = 'ServiceDescriptionItem';
