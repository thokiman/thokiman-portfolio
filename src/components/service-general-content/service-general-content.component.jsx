import React from "react";

import "./service-general-content.styles.scss";

const ServiceGeneralContent = ({ title, description }) => {
  return (
    <div className="service-general-content">
      <div className="service-title">{title}</div>
      <div className="service-description">
        {description.map(({ id, title, description }) => {
          return (
            <div key={id} className="service-description-container">
              <div className="service-description-title">{title}</div>
              <div className="service-description-item">{description}</div>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default ServiceGeneralContent;
