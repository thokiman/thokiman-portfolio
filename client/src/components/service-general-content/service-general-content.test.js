import React from "react";
import { shallow } from "enzyme";

import { ServiceGeneralContent } from "./service-general-content.component";

describe("ServiceGeneralContent component", () => {
  let wrapper;

  const mockText = "";
  const mockArray = [];
  beforeEach(() => {
    const mockProps = {
      title: mockText,
      description: mockArray,
    };
    wrapper = shallow(<ServiceGeneralContent {...mockProps} />);
  });

  it("should render ServiceGeneralContent component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
