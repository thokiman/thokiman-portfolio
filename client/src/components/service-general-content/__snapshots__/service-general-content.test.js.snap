// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceGeneralContent component should render ServiceGeneralContent component 1`] = `ShallowWrapper {}`;
